docs/testing/test-types/README.md

```diff
@@ -0,0 +1,46 @@
+# Test Types
+
+In the LocalStack codebase we differentiate between the following test types:
+
+- Unit tests
+- Acceptance Tests
+- Integration Tests
+
+Depending on the workflow and its trigger not all of those tests are executed at once.
+For ordinary pushes to `master` we only want to execute the Unit and Acceptance tests.
+On a regular schedule, however, we want to execute all tests to have as big of a coverage of our logic as possible.
+This differentiation also educates what we expect from the different types of tests.
+
+## Unit tests
+
+As the name suggests, these tests are performed on smaller units of logic to check if they're sound and perform the operations they claim to.
+This small unit can most often be a kind of helper function inside of a larger procedure.
+These tests should be able to complete their execution very quickly, so they never contain any interaction with some kind of infrastructure.
+If you need some kind of waiting mechanism in your unit test, it is most likely that you are not writing a unit test.
+
+
+## Acceptance tests
+
+We use acceptance tests to gain a quick understanding of whether the recently pushed commit to `master` fulfils minimally viable quality criteria.
+This means that these tests do not aim at maximum coverage but instead should test that the most important functionality works.
+This in general is the entire serving infrastructure and the main features of the most used services.
+
+As these tests are executed very often we need them to be as stable, fast and relevant as possible.
+We ensure this by the following criteria:
+
+- Should perfectly conform to all the testing rules laid out [here](../README.md)
+- Does not contain wait times (e.g., for resources to spin up) averaging longer than 20 seconds
+- Should be able to run in parallel. If tests need to run together (e.g., in a scenario), they need to be added to the same test class.
+- Displays some kind of real-world usage. This can be a scenario with multiple services. 
```

Comment:
I would move the `Displays some kind of real-world usage...` to the top of the list.
```suggestion
- Displays some kind of real-world usage. This can be a scenario with multiple services involved and tested. 
```